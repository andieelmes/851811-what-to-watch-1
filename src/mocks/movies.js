export const MOVIES = [
  {
    id: `fantastic-beasts-the-crimes-of-grindelwald`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
  },
  {
    id: `bohemian-rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`,
  },
  {
    id: `seven-years-in-tibet`,
    img: `img/seven-years-in-tibet.jpg`,
    title: `Seven Years in Tibet`,
  },
  {
    id: `what-we-do-in-the-shadows`,
    img: `img/what-we-do-in-the-shadows.jpg`,
    title: `What We Do in the Shadows`,
  },
  {
    id: `pulp-fiction`,
    img: `img/pulp-fiction.jpg`,
    title: `Pulp Fiction`,
  },
];
