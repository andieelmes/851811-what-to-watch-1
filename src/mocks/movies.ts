export const MOVIES = [
  {
    id: 1,
    img: 'https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg',
    title: 'Midnight Special',
    preview: 'https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4',
    video: 'http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4',
    genre: 'action',
    poster: 'https://es31-server.appspot.com/wtw/static/film/poster/Midnight_Special.jpg',
    backgroundImage: 'https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg',
    backgroundColor: '#828585',
    description: 'A father and son go on the run, pursued by the government and a cult drawn to the child\'s special powers.',
    rating: 3.3,
    ratingsCount: 67815,
    director: 'Jeff Nichols',
    starring: [
      'Michael Shannon',
      'Joel Edgerton',
      'Kirsten Dunst '
    ],
    duration: 112,
    year: 2016,
    favorite: false
  },
  {
    id: 2,
    img: 'https://es31-server.appspot.com/wtw/static/film/background/What-We-Do-in-the-Shadows.jpg',
    title: 'What We Do in the Shadows',
    preview: 'https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm',
    video: 'http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4',
    genre: 'comedy',
    poster: 'https://es31-server.appspot.com/wtw/static/film/poster/What-We-Do-in-the-Shadows.jpg',
    backgroundImage: 'https://es31-server.appspot.com/wtw/static/film/background/What-We-Do-in-the-Shadows.jpg',
    backgroundColor: '#A39E81',
    description: 'A look into the daily (or rather, nightly) lives of three vampires who\'ve lived together for over 100 years, in Staten Island.',
    rating: 4.2,
    ratingsCount: 6173,
    director: 'Jemaine Clement',
    starring: [
      'Kayvan Novak',
      'Matt Berry',
      'Natasia Demetriou'
    ],
    duration: 30,
    year: 2019,
    favorite: true
  }
];

export const GENRES = [
  `all genres`, `comedies`, `crime`, `documentaries`, `dramas`,
];

export const REVIEWS = [
  {
    id: 1,
    user: {
      id: 13,
      name: 'Zak'
    },
    rating: 2,
    comment: 'A movie that will take you to another world full of emotions.',
    date: '2019-05-30T04:04:34.909Z'
  },
  {
    id: 2,
    user: {
      id: 11,
      name: 'Jack'
    },
    rating: 5,
    comment: 'I really find it difficult to believe this movie is rated highly by people. It\'s hands down the worst movie I\'ve seen in my life',
    date: '2019-06-02T04:04:34.909Z'
  },
  {
    id: 3,
    user: {
      id: 14,
      name: 'Corey'
    },
    rating: 4,
    comment: 'It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.',
    date: '2019-06-10T04:04:34.909Z'
  }
];
