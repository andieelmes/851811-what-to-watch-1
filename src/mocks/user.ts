export const USER = {
  avatar: '/',
  authorized: true,
  name: 'name'
}
