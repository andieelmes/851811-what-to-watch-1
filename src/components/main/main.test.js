import React from 'react';
import renderer from 'react-test-renderer';
import Main from 'components/main/main.jsx';

import {MOVIES} from 'mocks/movies';

it(`Main component renders correctly`, () => {
  const tree = renderer
    .create(
        <Main
          movies={MOVIES}
        />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
