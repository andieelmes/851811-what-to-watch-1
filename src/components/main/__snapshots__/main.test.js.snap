// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render correctly for guests 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/sign-in"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Documentaries
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Dramas
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={1}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            style={null}
          >
            <img
              alt="Fantastic Beasts: The Crimes of Grindelwald"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={2}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            style={null}
          >
            <img
              alt="Bohemian Rhapsody"
              height="175"
              src="img/bohemian-rhapsody.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="img/bohemian-rhapsody.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={3}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            style={null}
          >
            <img
              alt="Seven Years in Tibet"
              height="175"
              src="img/seven-years-in-tibet.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="img/seven-years-in-tibet.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Seven Years in Tibet
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={4}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            style={null}
          >
            <img
              alt="What We Do in the Shadows"
              height="175"
              src="img/what-we-do-in-the-shadows.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="img/what-we-do-in-the-shadows.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={5}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            style={null}
          >
            <img
              alt="Pulp Fiction"
              height="175"
              src="img/pulp-fiction.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="img/pulp-fiction.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Pulp Fiction
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`Main component should render correctly when authorized 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar of name"
            height="63"
            src=""
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Documentaries
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Dramas
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={1}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            style={null}
          >
            <img
              alt="Fantastic Beasts: The Crimes of Grindelwald"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={2}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            style={null}
          >
            <img
              alt="Bohemian Rhapsody"
              height="175"
              src="img/bohemian-rhapsody.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="img/bohemian-rhapsody.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={3}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            style={null}
          >
            <img
              alt="Seven Years in Tibet"
              height="175"
              src="img/seven-years-in-tibet.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="img/seven-years-in-tibet.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Seven Years in Tibet
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={4}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            style={null}
          >
            <img
              alt="What We Do in the Shadows"
              height="175"
              src="img/what-we-do-in-the-shadows.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="img/what-we-do-in-the-shadows.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={5}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            style={null}
          >
            <img
              alt="Pulp Fiction"
              height="175"
              src="img/pulp-fiction.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="img/pulp-fiction.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Pulp Fiction
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
